panel/src/config/fiber.js

```diff
@@ -217,137 +405,81 @@ const Fiber = {
     }
   },
 
+  /**
+   * Updates the browser history and the page
+   * object with the current state
+   *
+   * @param {object} page
+   * @param {string} action
+   */
   state(page, action = "replace") {
     this.page = page;
     window.history[action + "State"](page, "", page.$url);
   },
 
-  toQuery(search, data) {
-    let params = new URLSearchParams(search);
-
-    // make sure that we always work with a data object
-    if (typeof data !== "object") {
-      data = {};
-    }
-
-    // add all data params unless they are empty/null
-    Object.entries(data).forEach(([key, value]) => {
-      if (value !== null) {
-        params.set(key, value);
-      }
-    });
-
-    return params;
-  },
-
-  toUrl(href, { data = {}, hash = true } = {}) {
-    let url;
-
-    if (hash === true) {
-      url = new URL(href, window.location);
+  /**
+   * Builds a full URL object based on the
+   * given path or another URL object and query data
+   *
+   * @param {string|URL} url
+   * @param {Object} query
+   * @returns
+   */
+  url(url = "", query = {}) {
+    if (typeof url === "string" && url.startsWith("http") === false) {
```

Comment:
What about a plugin view with a name that starts with `http`? Safer:

```suggestion
    if (typeof url === "string" && url.match(/^https?:\/\//) === null) {
```